akka-docs-dev/rst/scala/code/docs/stream/StreamTestKitDocSpec.scala

```diff
@@ -7,27 +7,99 @@ import akka.stream._
 import akka.stream.scaladsl._
 import akka.stream.testkit._
 import akka.stream.testkit.scaladsl._
+import scala.util._
+import scala.concurrent.duration._
+import scala.concurrent._
+import akka.testkit.TestProbe
+import akka.pattern.pipe
+import akka.pattern
 
 class StreamTestKitDocSpec extends AkkaSpec {
 
   implicit val mat = ActorFlowMaterializer()
 
-  "test source probe" in {
+  "strict collection" in {
+    //#strict-collection
+    val sinkUnderTest = Flow[Int].map(_ * 2).toMat(Sink.fold(0)(_ + _))(Keep.right)
 
-    //#test-source-probe
-    TestSource.probe[Int]
-      .toMat(Sink.cancelled)(Keep.left)
-      .run()
-      .expectCancellation()
-    //#test-source-probe
+    val future = Source(1 to 4).runWith(sinkUnderTest)
+    Await.ready(future, 100.millis)
+    val Success(result) = future.value.get
+    assert(result == 20)
```

Comment:
I used `val Success(result: Int) = future.value.get` to be in concert with the [testkit docs](http://doc.akka.io/docs/akka/2.3.11/scala/testing.html), but here I needed also to await until futures are completed. Now I see that I can simplify this with using `Await.result`.